src/components/LocationPage/ChartsHolder.tsx

```diff
@@ -44,13 +47,89 @@ const scrollTo = (div: null | HTMLDivElement, offset: number = 180) =>
     behavior: 'smooth',
   });
 
-interface ChartsHolderProps {
+interface RecommendationsProps {
   projections: Projections;
+  recommendationsRef: React.RefObject<HTMLDivElement>;
+}
+
+const Recommendations = ({
```

Comment:
I broke Recommendations out as a wrapper for the recommend thing because 1) it was almost entirely independent and 2) its props depended largely on projection data.   By splitting it, it's easier to conditionally load based on projection data being available.

Reply:
Is it important to break it out into a separate component file?  As it wasn't shared or used by anything else, it seemed okay to leave it here, but please let me know what you prefer.

Reply:
Thanks for adding all these useful comments! Separating the parts that depend on projections makes sense and makes the above the fold part load much faster compared to the rest (it feels almost instant in this branch), that's a huge improvement!

We are starting a redesign of the location page in a week or two, so we will have to implement/modify these blocks anyways, so don't worry about moving Recommendations to its own file, we can do that as part of the redesign. We will keep the structure of rendering what we can without projections for sure though!